scipy/stats/tests/test_stats.py

```diff
@@ -900,6 +900,56 @@ def test_index_error(self):
             0.0007835762419683435])
         assert_raises(ValueError, stats.spearmanr, x, y, axis=2)
 
+    def test_alternative(self):
+        # Test alternative parameter
+        assert_raises(ValueError, stats.spearmanr, X, X,
+                      alternative='error')
+
+        # Simple test - Based on the above ``test_spearmanr_vs_r``
+        x1 = [1, 2, 3, 4, 5]
+        x2 = [5, 6, 7, 8, 7]
+        expected = (0.82078268166812329, 0.088587005313543798)
+
+        res = stats.spearmanr(x1, x2, alternative="less")
+        assert_approx_equal(res[0], expected[0])
+        assert_approx_equal(res[1], 1 - (expected[1] / 2))
+
+        res = stats.spearmanr(x1, x2, alternative="greater")
+        assert_approx_equal(res[0], expected[0])
+        assert_approx_equal(res[1], expected[1] / 2)
+
+        # Test nan policies
+        x1nan = x1 + [np.nan]
+        x2nan = x2 + [np.nan]
+
+        assert_array_equal(stats.spearmanr(x1nan, x2nan), (np.nan, np.nan))
+
+        # Test omission
+        res_actual = stats.spearmanr(x1nan, x2nan, nan_policy='omit')
+        res_expected = stats.spearmanr(x1, x2)
+        assert_allclose(res_actual, res_expected)
+
+        res_actual = stats.spearmanr(x1nan, x2nan, nan_policy='omit',
+            alternative="less")
```

Comment:
Here and in a few more places below, adjust the indentation of the continued line:
```suggestion
        res_actual = stats.spearmanr(x1nan, x2nan, nan_policy='omit',
                                     alternative="less")
```